service/resource/cloudformationv2/desired.go

```diff
@@ -0,0 +1,23 @@
+package cloudformationv2
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/keyv2"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
+	customObject, err := keyv2.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	mainStack, err := newMainStack(customObject)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
```

Comment:
Also here I'd expect the desired state to include all 3 stacks.

Reply:
Yep, let's discuss in the related issue, will link here when created

Reply:
Here it is https://github.com/giantswarm/aws-operator/issues/621